File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/models/Teacher.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlacementCellBackend.Models;

public abstract class CollegeStaff
{
    [Key]
    public string Id { get; set; } = string.Empty;

    [Required]
    public int CollegeId { get; set; }

    [ForeignKey("CollegeId")]
    public College? College { get; set; }

    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
}

public class Teacher : CollegeStaff
{
    public string Department { get; set; } = string.Empty;
}

public class PlacementOfficer : CollegeStaff
{
    public string PhoneNo { get; set; } = string.Empty;
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/models/Teacher.java
package com.example.placementicsbackend.models;

import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;
import java.util.UUID;
@Entity
@Table(name = "teacher", uniqueConstraints = @UniqueConstraint(name = "uq_teacher_email", columnNames = "email"))
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Teacher {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "college_id", nullable = false)
    private College college;
    @Column(nullable = false, length = 255)
    private String name;
    @Column(nullable = false, length = 255)
    private String email;
    @Column(length = 150)
    private String department;
    @Column(name = "joining_year")
    private Integer joiningYear;
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
    @OneToOne(mappedBy = "teacher")
    private UserAccount userAccount;
    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
    }
    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}